// @civitics/db/src/types/database.ts
// This file will be auto-generated by Supabase CLI:
//   pnpm --filter @civitics/db db:types
//
// Placeholder until local Supabase instance is running.

export type Json =
  | string
  | number
  | boolean
  | null
  | { [key: string]: Json | undefined }
  | Json[];

export interface Database {
  public: {
    Tables: {
      jurisdictions: {
        Row: {
          id: string;
          parent_id: string | null;
          type: string;
          name: string;
          short_name: string | null;
          country_code: string | null;
          fips_code: string | null;
          census_geoid: string | null;
          population: number | null;
          timezone: string | null;
          is_active: boolean;
          metadata: Json;
          created_at: string;
          updated_at: string;
        };
        Insert: Omit<Database["public"]["Tables"]["jurisdictions"]["Row"], "id" | "created_at" | "updated_at"> & {
          id?: string;
        };
        Update: Partial<Database["public"]["Tables"]["jurisdictions"]["Insert"]>;
      };
      officials: {
        Row: {
          id: string;
          governing_body_id: string;
          jurisdiction_id: string;
          full_name: string;
          first_name: string | null;
          last_name: string | null;
          role_title: string;
          party: string | null;
          photo_url: string | null;
          email: string | null;
          website_url: string | null;
          district_name: string | null;
          term_start: string | null;
          term_end: string | null;
          is_active: boolean;
          is_verified: boolean;
          source_ids: Json;
          metadata: Json;
          created_at: string;
          updated_at: string;
        };
        Insert: Omit<Database["public"]["Tables"]["officials"]["Row"], "id" | "created_at" | "updated_at"> & {
          id?: string;
        };
        Update: Partial<Database["public"]["Tables"]["officials"]["Insert"]>;
      };
      proposals: {
        Row: {
          id: string;
          governing_body_id: string | null;
          jurisdiction_id: string;
          type: string;
          status: string;
          title: string;
          short_title: string | null;
          bill_number: string | null;
          introduced_at: string | null;
          comment_period_start: string | null;
          comment_period_end: string | null;
          regulations_gov_id: string | null;
          summary_plain: string | null;
          summary_generated_at: string | null;
          source_ids: Json;
          metadata: Json;
          created_at: string;
          updated_at: string;
        };
        Insert: Omit<Database["public"]["Tables"]["proposals"]["Row"], "id" | "created_at" | "updated_at"> & {
          id?: string;
        };
        Update: Partial<Database["public"]["Tables"]["proposals"]["Insert"]>;
      };
      users: {
        Row: {
          id: string;
          privy_user_id: string | null;
          email: string | null;
          display_name: string | null;
          avatar_url: string | null;
          district_jurisdiction_id: string | null;
          zip_code: string | null;
          civic_credits_balance: number;
          wallet_address: string | null;
          is_email_verified: boolean;
          is_government_verified: boolean;
          ai_queries_today: number;
          ai_queries_reset_at: string | null;
          is_active: boolean;
          created_at: string;
          updated_at: string;
        };
        Insert: Omit<Database["public"]["Tables"]["users"]["Row"], "id" | "created_at" | "updated_at"> & {
          id?: string;
        };
        Update: Partial<Database["public"]["Tables"]["users"]["Insert"]>;
      };
    };
    Views: Record<string, never>;
    Functions: Record<string, never>;
    Enums: {
      jurisdiction_type: "global" | "supranational" | "country" | "state" | "county" | "city" | "district" | "precinct" | "other";
      proposal_type: "bill" | "resolution" | "amendment" | "regulation" | "executive_order" | "treaty" | "referendum" | "initiative" | "budget" | "appointment" | "ordinance" | "other";
      proposal_status: "introduced" | "in_committee" | "passed_committee" | "floor_vote" | "passed_chamber" | "passed_both_chambers" | "signed" | "vetoed" | "veto_overridden" | "enacted" | "open_comment" | "comment_closed" | "final_rule" | "failed" | "withdrawn" | "tabled";
      party: "democrat" | "republican" | "independent" | "libertarian" | "green" | "other" | "nonpartisan";
      connection_type: "donation" | "vote_yes" | "vote_no" | "vote_abstain" | "appointment" | "revolving_door" | "oversight" | "lobbying" | "co_sponsorship" | "family" | "business_partner" | "legal_representation" | "endorsement" | "contract_award";
      promise_status: "made" | "in_progress" | "kept" | "broken" | "partially_kept" | "expired" | "modified";
    };
  };
}
